clients/cobol-lsp-vscode-extension/src/task/CompileTaskProvider.ts

```diff
@@ -0,0 +1,75 @@
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import * as vscode from "vscode";
+import {taskType} from "../extension";
+import { CustomBuildTaskTerminal } from "./CustomBuildTaskTerminal";
+
+const TASK_DETAIL = "compile with IBM Enterprise COBOL for z/OS";
+export class CompileTaskProvider implements vscode.TaskProvider {
+    public static taskSource = "cobol";
+    public static problemMatcher = "$enterprise-cobol";
+    public static taskProviderName = "compile";
+    private cobolCompileTasks?: Thenable<vscode.Task[]> = undefined;
+
+    public resolveTask(task: vscode.Task, _token: vscode.CancellationToken): vscode.ProviderResult<vscode.Task> {
+        const definition: CobolCompileJobDefinition = <any> task.definition;
+        if (definition.jobCard) {
+                // resolveTask requires that the same definition object be used.
+                const documentText = vscode.window.activeTextEditor.document;
+                return new vscode.Task(
+                    definition,
+                    vscode.TaskScope.Workspace,
+                    CompileTaskProvider.taskProviderName,
+                    CompileTaskProvider.taskSource,
+                    new vscode.CustomExecution(async () => new CustomBuildTaskTerminal(documentText, definition)),
+                    CompileTaskProvider.problemMatcher,
+                );
+            }
+        return undefined;
+    }
+
+    public provideTasks(_token: vscode.CancellationToken): vscode.ProviderResult<vscode.Task[]> {
+        if (!this.cobolCompileTasks) {
+            this.cobolCompileTasks = this.getCompileTask();
+        }
+        return this.cobolCompileTasks;
+    }
+
+    private async getCompileTask(): Promise<vscode.Task[]> {
+        const result: vscode.Task[] = [];
+        const documentText = undefined;
+        const definition: CobolCompileJobDefinition = {
+            type: taskType,
+            jobCard: [],
+        };
+        const task = new vscode.Task(definition, vscode.TaskScope.Workspace, CompileTaskProvider.taskProviderName,
+            CompileTaskProvider.taskSource,
+            new vscode.CustomExecution(async () => new CustomBuildTaskTerminal(documentText, definition)),
+            CompileTaskProvider.problemMatcher);
+        task.detail = TASK_DETAIL;
+        result.push(task);
+        return result;
+    }
+}
+
+export interface CobolCompileJobDefinition extends vscode.TaskDefinition {
+    jobCard: string[];
+    steplib?: string[];
+    syslib?: string[];
+    saveListing?: boolean;
+    listingLocation?: string;
+    compilerOptions?: string[];
+    unit?: string;
+}
```

Comment:
How do we make sure this type is in sync with the contribution point in package.json?

Reply:
Manual as of now.